### PR TITLE
Adding packets EVC 22 to EVC-30

### PR DESCRIPTION
**Pull request by robertvanv**
I have added the next batch of packets EVC-22 up to EVC-30

There are a few TODOs that I would like you to check to make sure that the NestedDataSets are correct, as well as the static bitfield implementations that I created. Also, there is a question about InvertBits in there as well under a TODO.

Let me know what you think.

### CODE REVIEW

File: Parsers/VSISDMI.cs [hunk 1]
```diff
@@ -1666,14 +1656,19 @@ public VSISDMI()
                                 Name = "MMI_Q_LEVEL_NTC_ID",
                                 BitFieldType = BitFieldType.Bool,
                                 Length = 1,
-                                Comment = "Qualifier for the variable MMI_M_LEVEL_NTC_ID"
+                                Comment = "Qualifier for the variable MMI_M_LEVEL_NTC_ID",
```
Comment (dyster) [inline Parsers/VSISDMI.cs:1659]: As discussed on skype, Bool does not match with "0" or "1"

File: Parsers/VSISDMI.cs [hunk 2]
```diff
@@ -1724,296 +1728,1124 @@ public VSISDMI()
...
+                            {
+                                VariableLengthSettings = new VariableLengthSettings
+                                {
+                                    Name = "MMI_N_ITER2"
+                                },
+                                NestedDataSet = new DataSetDefinition
+                                {
+                                    BitFields = new List<BitField>
+                                    {
+                                        MMI_STM_L_VALUE2,
```
Comment (dyster) [inline Parsers/VSISDMI.cs:2054]: The definition of this is the same as MMI_STM_L_VALUE so they should be merged

File: Parsers/VSISDMI.cs [hunk 3]
```diff
@@ -1724,296 +1728,1124 @@ public VSISDMI()
...
+                "230530240",
+                "230535240"
+            },
+            BitFields = new List<BitField>
+            {
+                MMI_M_PACKET,
+                MMI_L_PACKET,
+                new BitField
+                {
+                    // TODO How do you set just this BitField with the InvertBits parameter?
```
Comment (dyster) [inline Parsers/VSISDMI.cs]: make a nested dataset and set InvertBits on it

File: Parsers/VSISDMI.cs [hunk 4]
```diff
@@ -2282,6 +3132,80 @@ public VSISDMI()
...
+
+        public static BitField MMI_STM_Q_FOLLOWING => new BitField
+        {
+            Name = "MMI_STM_Q_FOLLOWING",
+            BitFieldType = BitFieldType.Bool,
+            Length = 1,
+            Comment = "Indicate following data to be viewed at the same time"
+        };
+
+        // TODO Johan, please check whether the length is required along with the scaling factor
```
Comment (dyster) [inline Parsers/VSISDMI.cs:3207]: No the length will be ignored when VariableLength is set. You can look in BitStreamParser.cs in s-tools to see the logic

File: Parsers/VSISDMI.cs [hunk 5]
```diff
@@ -2282,6 +3132,80 @@ public VSISDMI()
...
+            Comment = "Data value caption text byte string",
+            VariableLengthSettings = new VariableLengthSettings
+            {
+                Name = "MMI_STM_L_VALUE",
+                ScalingFactor = 8
+            }
+        };
+
+        // TODO Johan, please check whether the length is required along with the scaling factor
+        public static BitField MMI_STM_X_VALUE2 => new BitField
```
Comment (dyster) [inline Parsers/VSISDMI.cs:3236]: This one looks suspiciously similar to the previous one

**Review by dyster — CHANGES_REQUESTED** [review]
See comments in code

File: Parsers/VSISDMI.cs (hunk 2 above)
Comment (robertvanv) [reply Parsers/VSISDMI.cs:2054]: I thought about merging them but I noticed that the comments are slightly different (as per VSIS). I agree the differences are minimal so we can merge them if you like.

MMI_STM_L_VALUE Comment = "Length of X_VALUE for default value (range: 0-10)."
MMI_STM_L_VALUE2 Comment = "Length of X_VALUE for pick-up list value (range: 0-10)."

File: Parsers/VSISDMI.cs (hunk 4 above)
Comment (robertvanv) [reply Parsers/VSISDMI.cs:3207]: Oh yes, i remember now. I guess best to leave it in for clarity though.

File: Parsers/VSISDMI.cs (hunk 5 above)
Comment (robertvanv) [reply Parsers/VSISDMI.cs:3236]: Yes, they are the same, but VSIS refers to them as different variables. I am happy to reuse MMI_STM_X_VALUE, but just keep in mind that it won't exactly align with the VSIS structure.

File: Parsers/VSISDMI.cs (hunk 1 above)
Comment (robertvanv) [reply Parsers/VSISDMI.cs:1659]: Ok, I will change it to UInt8

File: Parsers/VSISDMI.cs (hunk 3 above)
Comment (robertvanv) [reply Parsers/VSISDMI.cs]: Nested DataSet implemented

**Review by robertvanv — COMMENTED** [review]
Attempted to change code on GitHub, along with comments

**Review by dyster — APPROVED** [review]

**Pull request merged**